rust/rope/src/tree.rs

```diff
@@ -116,29 +123,48 @@ enum NodeVal<N: NodeInfo> {
 
 // also consider making Metric a newtype for usize, so type system can
 // help separate metrics
+
+/// A trait for quickly processing attributes of a NodeInfo.
+/// 
+/// For the conceptual background see the 
+/// [blog post, Rope science, part 2: metrics](https://github.com/google/xi-editor/blob/master/doc/rope_science/rope_science_02.md).
 pub trait Metric<N: NodeInfo> {
+    /// Return the number of boundarys in the NodeInfo::Leaf
+    /// 
+    /// TODO: What does the second argument do?
     fn measure(&N, usize) -> usize;
 
+    /// Return a offset in base units forwich self.from_base_units will return the argument.
+    /// 
+    /// TODO: Does it have to be the largest, smallest, or just eny offset?
     fn to_base_units(l: &N::L, in_measured_units: usize) -> usize;
 
+    /// Return the number of boundarys in the NodeInfo::Leaf befor the offset in_base_units.
```

Comment:
"Returns the offset in measured units corresponding to an offset in base units."